backend/consensus/base.py

```diff
@@ -202,20 +218,44 @@ async def exec_transaction(
 
             num_validators = len(remaining_validators) + 1
 
-            contract_snapshot = contract_snapshot_factory(transaction.to_address)
+            contract_snapshot_supplier = lambda: contract_snapshot_factory(
+                transaction.to_address
+            )
+
+            leaders_contract_snapshot = contract_snapshot_supplier()
 
             # Create Leader
             leader_node = node_factory(
                 leader,
                 ExecutionMode.LEADER,
-                contract_snapshot,
+                leaders_contract_snapshot,
                 None,
                 msg_handler,
                 contract_snapshot_factory,
             )
 
             # Leader executes transaction
             leader_receipt = await leader_node.exec_transaction(transaction)
+
+            if leader_receipt.error is not None:
+                # FIXME
```

Comment:
what's this?

Reply:
Just removed and it appears that if there is only one validator then deploying a contract that errors in the constructor succeeds

Reply:
I agree that this solution is bad and fixme was left on purpose =)
but we probably need to find a better solution before merging

Reply:
I came to a conclusion that here we should ceil round and not floor round
`num_agree >= num_validators // 2`
=>
`num_agree  >= (num_validators + 1) // 2`

Reply:
removed in 9539ce042d8c775ca019017333961e215ead1dfb

Reply:
![image](https://github.com/user-attachments/assets/c72246f7-b970-49d3-ad47-7b4691c33f18)

And "undeterminated" spins in transactions list...